[Do Not Merge] Pytorch components trainer

ankan94 opened the pull request:
**Description of your changes:**


**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
<!--
   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.
-->

chensun committed:
chore: restore KFP SDK version to 1.6.2, decouple SDK release from KFP main release. (#5739)

* Update __init__.py

* Update bump-version-in-place.sh

juliusvonkohout committed:
Fix the cache-deployer to run as non root (#5742)

shrinath-suresh committed:
Adding PyTorch training component

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

shrinath-suresh committed:
Adding PyTorch - trainer unit tests

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

sasha-gitg committed:
fix(components/google-cloud): update version to include google-cloud-aiplatform fix (#5721)

* fix: update version to include google-cloud-aiplatform fix

* fix: update dockerfile to build from PR, will update once google-cloud-aiplatform releases fix

* fix: upgrade dependency to latest google-cloud-aiplatform release

chensun committed:
chore: Adding chensun in KFP root OWNERS (#5747)

dependabot[bot] committed:
chore(deps): bump dns-packet from 1.3.1 to 1.3.4 in /frontend (#5749)

Bumps [dns-packet](https://github.com/mafintosh/dns-packet) from 1.3.1 to 1.3.4.
- [Release notes](https://github.com/mafintosh/dns-packet/releases)
- [Changelog](https://github.com/mafintosh/dns-packet/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mafintosh/dns-packet/compare/v1.3.1...v1.3.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ankan94 committed:
Fixing UT and adding setup.py

Signed-off-by: ankan94 <ankan@ideas2it.com>

chensun committed:
fix(sdk.v2): Fix the bug where string concat inside a loop may break compilation. (#5752)

chensun committed:
fix(sdk): Avoid inserting duplicate input parameters in v2 compatible mode. (#5740)

Ark-kun committed:
feat(sdk): Dropped support for Python 3.5. Fixes #4584 (#4611)

Fixes https://github.com/kubeflow/pipelines/issues/4584

kbc8894 committed:
fix(cli): handle when pipeline.paramters is None (#5725)

Signed-off-by: ByungchanKim <kbc8894@gmail.com>

Arvind-Ideas2IT committed:
update setup.py, trainer compopents pyfiles,applied pylyintrc

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

chauhang commented on components/PyTorch/pytorch-kfp-components/setup.py:
```diff
@@ -0,0 +1,65 @@
+    name="pytorch-kfp-components",
+    version=version,
+    description="PyTorch Kubeflow Pipeline",
+    url="https://github.com/chauhang/pipelines/tree/master/components",
```
This will be the url of the upstream location

chauhang commented on components/PyTorch/pytorch-kfp-components/setup.py:
```diff
@@ -0,0 +1,65 @@
+        "torch",
+	    "kfp",
+        "torchserve",
+	    "pytorch-lightning"
```
Do we need to tag a version for Lightning?

chauhang commented on components/PyTorch/pytorch-kfp-components/setup.py:
```diff
@@ -0,0 +1,65 @@
+    url="https://github.com/chauhang/pipelines/tree/master/components",
+    author="The PyTorch Pipeline Components authors",
+    author_email="PyTorch-pipeline-components@fb.com",
+    license="Apache License 2.0",
```
To be replaced with BSD License

chauhang commented on components/PyTorch/pytorch-kfp-components/LICENSE:
```diff
@@ -0,0 +1,201 @@
+				Apache License
```
Replace with BSD License, use this as reference and just change the "For xxx" line https://github.com/pytorch/tensorpipe/blob/master/LICENSE.txt

chauhang commented on components/PyTorch/tests/iris/iris_data_module.py:
```diff
@@ -0,0 +1,101 @@
+from argparse import ArgumentParser
```
Add the copyright headers

chauhang commented on components/PyTorch/tests/iris/iris_classification.py:
```diff
@@ -0,0 +1,84 @@
+# pylint: disable=W0221
```
Add copyright headers for this and other files

chauhang commented on components/PyTorch/tests/iris/iris_pytorch.py:
```diff
@@ -0,0 +1,53 @@
+from argparse import ArgumentParser
```
Add copyright header

chauhang commented on components/PyTorch/tests/iris/iris_handler.py:
```diff
@@ -0,0 +1,53 @@
+import ast
```
Add copyright header

chauhang reviewed (changes requested):
Please see comments inline

chensun committed:
feat(sdk): Better error message when compiling for v2. Fixes #5688, Fixes #5727 (#5759)

* Better error message when compiling for v2

* update samples

wakanapo committed:
fix(sdk, v2): Fix error when replica_count isn't set (#5761)

Bobgy committed:
chore: set up AUTHORS file. Fixes #5470 (#5766)

Arvind-Ideas2IT committed:
adressed the review comments,applied pylint, black

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

Arvind-Ideas2IT committed:
added copyright headers in tests files

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

Arvind-Ideas2IT commented:
> Please see comments inline

Hi @chauhang, I have addressed your comments for setup.py. and for copyright headers

Arvind-Ideas2IT committed:
updated base component

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

zijianjoy committed:
feat(frontend): MetricsTab for NodeDetail, ROC Curve from MLMD custom property. Issue #5668 (#5731)

* feat(frontend): MetricsTab for NodeDetail, ROC Curve from MLMD custom property

* add back test renderer

* remove debug tool

* address nit comments

* more comment address

* merge metrics to visualization

* error boundary, revert banner change, more testing, data validation

* attempt to use actual function

* clear queryclient

* address comments

* address comments

* address comments

Bobgy committed:
feat(v2): enable v2 compatible mode in full Kubeflow with zero config. Fixes #5680 (#5697)

* feat(v2): enable v2 compatible mode in full Kubeflow

* fix

* fix

* move objectstore reelated code to separate package

* fix style

dependabot[bot] committed:
chore(deps): bump urllib3 from 1.26.3 to 1.26.5 in /sdk/python (#5769)

Bumps [urllib3](https://github.com/urllib3/urllib3) from 1.26.3 to 1.26.5.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/1.26.3...1.26.5)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump ws from 6.2.1 to 6.2.2 in /frontend/server (#5771)

Bumps [ws](https://github.com/websockets/ws) from 6.2.1 to 6.2.2.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/commits)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Bobgy committed:
test(samples): fix ai_platform.ipynb. Fixes #5775 (#5776)

Bobgy committed:
fix(deployment): GCP AI Platform Pipelines -- correct default values when some fields left empty. Fixes #5717 (#5777)

* fix(deployment): GCP AI Platform Pipelines -- correct default values when some fields left empty

* backward compatible databaseNamePrefix behavior

Bobgy committed:
feat(deployment): configurable v2 compatible mode default pipeline root. Part of #5680. Fixes #5704 (#5750)

* feat(deployment): configurable v2 compatible mode default pipeline root

* clarify documentation

chensun committed:
release(api): Bump kfp-pipeline-spec version to 0.1.8 (#5778)

chensun committed:
fix(sdk.v2): fix metrics outputs populating to unblock pipeline run (#5770)

chensun committed:
fix(sdk): resolve pipeline_root and run_uid placeholders in container env (#5772)

dependabot[bot] committed:
chore(deps): bump lodash from 4.17.19 to 4.17.21 in /frontend (#5624)

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.19 to 4.17.21.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.19...4.17.21)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
fix(sdk): Add http status code check for token requests. Fixes #4312 (#5782)

Signed-off-by: Anna Jung (VMware) <antheaj@vmware.com>

Bobgy committed:
feat: update MLMD to 1.0.0 (#5786)

* feat: update MLMD to 1.0.0

* update MLMD frontend client

zijianjoy committed:
feat(frontend): Support v2 confusion matrix visualization. Issue #5668 (#5781)

* feat(frontend): Shows ConfusionMatrix in Metrics Tab

* adjust uI

* remove comment

* definition description

* moving wrapper

* runtype check

chensun committed:
feat(sdk.v2): Support Exit handler in v2 compiler. (#5784)

ji-yaqi committed:
release: KFP SDK and v2 launcher 1.6.3 (#5796)

* Update __init__.py to 1.6.3

* Update v2_compat.py

hilcj committed:
Update pipeline_spec.proto (#5787)

The single quote is no longer supported.

ji-yaqi committed:
chore: update python version for build.sh (#5799)

jli committed:
chore: add link and Python 3.8/3.9 classifiers to setup.py (#5715)

* Add link and Python 3.8/3.9 classifiers to setup.py

The link should appear on PyPI: https://pypi.org/project/kfp/#description

* use "The Kubeflow Authors" in setup.py

* Include more links in Python package description

Bobgy committed:
chore(launcher): put pipeline context as parent of run context. Part of #5669 (#5785)

chauhang committed:
Changing to Apache2 License

Reviewed license change with Legal team

Commit:
Switch to Apache2 License

Commit:
Add tox for test automation with presubmit script

Commit:
Add model archiver to dependencies

Commit:
Cleanup setup.py

chauhang committed:
Merge branch 'kubeflow:master' into pytorch-components-trainer

Commit:
Cleanup

Commit:
Cleanup

Commit:
Cleanup

shrinath-suresh committed:
Using common fixture for unit tests and adding gpu fix for saving the model

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

Arvind-Ideas2IT committed:
Changing to Apache2 License

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

Windfarer committed:
fix(scheduledworkflow): Set location to make CRON timezone work, Fixes #2653 (#5800)

* set location

* update log

* fix log

* install tzdata

Arvind-Ideas2IT committed:
resolve typo

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

Commit:
Added OWNERS

Commit:
Merge branch 'pytorch-components-trainer' of https://github.com/chauhang/pipelines into pytorch-components-trainer

Commit:
OWENRS will be added as separate PR

Arvind-Ideas2IT committed:
feat(components): Adding Mar component for PyTorch - KFP (#5780)

* adding mar component

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

* removed extra files

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

* removed exta file __init__py

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

* Switch to Apache2 License

* Move tests folder

* Adding spec for mar component

Signed-off-by: Shrinath Suresh <shrinath@ideas2it.com>

Co-authored-by: Geeta Chauhan <gchauhan@fb.com>
Co-authored-by: Shrinath Suresh <shrinath@ideas2it.com>

Arvind-Ideas2IT committed:
fix detect_version function in setup.py

Signed-off-by: Arvind-Ideas2IT <arvindkumarsingh.gautam@ideas2it.com>

Commit:
Cleanup for setup description field warnings

shrinath-suresh committed:
Merge remote-tracking branch 'upstream/master' into pytorch-components-trainer
